PaulasWeatherApp/PaulasWeatherApp/ViewController/WeatherReportsViewController.swift

```diff
@@ -8,9 +8,102 @@
 import UIKit
 
 class WeatherReportsViewController: UIViewController {
+    
+    private var hourlyWeather: [WeatherDetails] = []
+    private let tableView = UITableView()
+    private var refresher = UIRefreshControl()
+    private let backgroundImage = UIImageView()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
-        view.backgroundColor = .systemYellow
+        setBackgroundImage()
+        setTableView()
+        setRefresher()
+        refresh()
+    }
+    
+    private func setRefresher() {
+        refresher.attributedTitle = NSAttributedString(string: "pull to refresh")
+        refresher.addTarget(self, action: #selector(WeatherReportsViewController.refresh), for: UIControl.Event.valueChanged)
+        self.tableView.addSubview(refresher)
+    }
+    
+    private func setTableView() {
+        view.addSubview(tableView)
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.backgroundColor = .clear
+    }
+    
+    private func setBackgroundImage() {
+        view.addSubview(backgroundImage)
+        backgroundImage.image = UIImage(named: "background")
+        backgroundImage.backgroundColor = .clear
+        backgroundImage.translatesAutoresizingMaskIntoConstraints = false
+        backgroundImage.bounds = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
+        
+        NSLayoutConstraint.activate([
+            backgroundImage.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            backgroundImage.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            backgroundImage.topAnchor.constraint(equalTo: view.topAnchor),
+            backgroundImage.bottomAnchor.constraint(equalTo: view.bottomAnchor) ])
+    }
+    
+    func setHourlyData( hourlyWeather: [WeatherDetails] ) {
+        let date = Date()
+        let calendar = Calendar.current
+        let hour = calendar.component(.hour, from: date)
+        
+        for (_, item) in hourlyWeather.enumerated() {
```

Comment:
if you don't need the index you can do just 
for item in  hourlyWeather.enumerated()